### PR TITLE
feat: add backwards compatibility for sheetId

### PR DESCRIPTION
**Pull request by setchy**
Support both documentId with sheetId for backwards compatibility

### CODE REVIEW

**Comment by dhair-dml01** [comment]
Does this change address the following constraint under the "Building the radar" heading in the readme?  

"Note: When using the BYOR app on [radar.thoughtworks.com](https://radar.thoughtworks.com/), the ring and quadrant names should be among the values mentioned in the [example above](https://github.com/setchy/build-your-own-radar/tree/feature/doc-sheet-backwards-compatibility#setting-up-your-data). This holds good for Google Sheet, CSV or JSON inputs. For a self hosted BYOR app, there is no such condition on the names. Instructions to specify custom names are in the [next section](https://github.com/setchy/build-your-own-radar/tree/feature/doc-sheet-backwards-compatibility#more-complex-usage)."

I have overridden ring and quadrant names in my Google sheet that no longer worked after the recent change.  Restoring that capability would be greatly appreciated.  Thanks.

**Comment by marisahoenig** [comment]
Hey @setchy — I'd love to get this merged in. Would you first be able to add a unit test to make sure the docId works? The ones testing the sheetId are in `sheet-spec.js`.

**Comment by setchy** [comment]
@marisahoenig - unit tests added 😄

File: spec/util/urlUtils-spec.js [hunk 1]
```diff
@@ -32,4 +32,40 @@ describe('Url Utils', () => {
     expect(config).toHaveBeenCalledTimes(1)
     expect(queryParams).toHaveBeenCalledTimes(1)
   })
+
+  it('should prioritize documentId before legacy sheetId', () => {
```
Comment (marisahoenig) [inline spec/util/urlUtils-spec.js:36]: Great idea for this unit test!

File: src/util/factory.js [hunk 2]
```diff
@@ -538,15 +539,10 @@ function plotUnauthorizedErrorMessage() {
...
-    if (featureToggles.UIRefresh2022) {
-      const queryString = window.location.href.match(/documentId(.*)/)
-      const queryParams = queryString ? QueryParams(queryString[0]) : {}
-      sheet = GoogleSheet(queryParams.documentId, queryParams.sheetName)
-    } else {
-      const queryString = window.location.href.match(/sheetId(.*)/)
-      const queryParams = queryString ? QueryParams(queryString[0]) : {}
-      sheet = GoogleSheet(queryParams.sheetId, queryParams.sheetName)
-    }
+    const queryString = window.location.href.match(/sheetId(.*)/)
```
Comment (marisahoenig) [inline src/util/factory.js:542]: Love the simplification

**Review by marisahoenig — COMMENTED** [review]
I think the changes are great. I sent a message to Devansh to just double check this — will get this merged in next week.

**Comment by devansh-sharma-tw** [comment]
@setchy , can you pls add the regex check for documentId as well [here](https://github.com/thoughtworks/build-your-own-radar/pull/318/files#diff-316c95c871c5d0350e6c3f1e4ea1477a63b1714bd756f03b033f5dafc566f021R539) in `factory.js` line 539 (similar to the change in urlUtils) ? Unless it was skipped for some specific case ?

Currently, this partially breaks the flow of Switching Accounts when accessing a private Google Sheet with the wrong user.

Thanks!

**Comment by setchy** [comment]
> @setchy , can you pls add the regex check for documentId as well [here](https://github.com/thoughtworks/build-your-own-radar/pull/318/files#diff-316c95c871c5d0350e6c3f1e4ea1477a63b1714bd756f03b033f5dafc566f021R539) in `factory.js` line 539 (similar to the change in urlUtils) ? Unless it was skipped for some specific case ?
> 
> Currently, this partially breaks the flow of Switching Accounts when accessing a private Google Sheet with the wrong user.
> 
> Thanks!

Good pick up @devansh-sharma-tw 👏.  Not sure how best to validate locally, but I've pushed a change that will use the sheetName function now

**Comment by devansh-sharma-tw** [comment]
@setchy `getSheetName` does work now.
Although I think you have pushed some other changes in `src/graphing/components/quadrantTables.js` and `src/graphing/radar.js` not related to this PR. If you can clean those up, we should be good to merge this :)

> Not sure how best to validate locally

One way to validate this locally (which I use) with 2 Google accounts is like this:
- Create a private Google Sheet with multiple sheets with Account A
- Open it in BYOR with a specific sheet (so URL would be like `localhost/?documentId=<Account A private sheet>&sheetName=<some sheet name>`)
- Login with Account B
- It should take you to the "Switch Account" screen
- Then logging in correctly with Account B should render the radar with the correct sheet (not the default sheet)

**Comment by setchy** [comment]
Oops 🙈.  All cleaned up now 🧼

**Comment by devansh-sharma-tw** [comment]
Looks good to merge now. Thanks!

**Pull request merged**